src/io/matpower.jl

```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+                    "\t", (-dcline["qf"]*mvabase), # opposite convention
+                    "\t", (-dcline["qt"]*mvabase), # opposite convention
+                    "\t", dcline["vf"],
+                    "\t", dcline["vt"],
+                    "\t", (dcline["pmin"]),
+                    "\t", (dcline["pmax"]),
+                    "\t", (dcline["qminf"] * mvabase),
+                    "\t", (dcline["qmaxf"] * mvabase),
+                    "\t", (dcline["qmint"] * mvabase),
+                    "\t", (dcline["qmaxt"] * mvabase),
+                    "\t", (dcline["loss0"] * mvabase),
+                    "\t", dcline["loss1"],
+                    "\t", (haskey(dcline, "mu_pmin") ? (dcline["mu_pmin"]*mvabase) : ""),
+                    "\t", (haskey(dcline, "mu_pmax") ? (dcline["mu_pmax"]*mvabase) : ""),
+                    "\t", (haskey(dcline, "mu_qminf") ? (dcline["mu_qminf"]*mvabase) : ""),
+                    "\t", (haskey(dcline, "mu_qmaxf") ? (dcline["mu_qmaxf"]*mvabase) : ""),
+                    "\t", (haskey(dcline, "mu_qmint") ? (dcline["mu_qmint"]*mvabase) : ""),
+                    "\t", (haskey(dcline, "mu_qmaxt") ? (dcline["mu_qmaxt"]*mvabase) : ""),
+        )
+    end     
+    println(io, "];")
+    println(io)
+           
+    # Print the gen cost data
+    if haskey(collect(values(generators))[1], "cost")
+        println(io, "%%-----  OPF Data  -----%%")
+        println(io, "%% generator cost data")
+        println(io, "%    1    startup    shutdown    n    x1    y1    ...    xn    yn")
+        println(io, "%    2    startup    shutdown    n    c(n-1)    ...    c0")
+        println(io, "mpc.gencost = [")
+        
+        
+        for (idx,gen) in (sort(generators))
+            if gen["model"] == 1
+                print(io, "\t1\t", gen["startup"], "\t", gen["shutdown"], "\t", (length(gen["cost"]) / 2) ),
+                for l=1:length(gen["cost"])
+                    print(io, "\t", gen["cost"][l] * mvabase^(l % 2))  
+                end
+            else                          
+                print(io, "\t2\t", gen["startup"], "\t", gen["shutdown"], "\t", length(gen["cost"])),
+                for l=1:length(gen["cost"])
+                    print(io, "\t", gen["cost"][l] / mvabase^(length(gen["cost"])-l))  
+                end
+            end
+            println(io)   
+        end
+        println(io, "];");
+        println(io)
+    end
+    
+    
+    
+    
+    
+    
+    
+    # Print the dcline cost data
+    if length(data["dcline"]) > 0
+        if haskey(collect(values(dclines))[1], "cost")
+            println(io, "%% dcline cost data")
+            println(io, "%    1    startup    shutdown    n    x1    y1    ...    xn    yn")
+            println(io, "%    2    startup    shutdown    n    c(n-1)    ...    c0")
+            println(io, "mpc.dclinecost = [")
+        
+            for (idx,dcline) in (sort(dclines))
```

Comment:
This block will always be identical to the generator one, might want to make it a function.